From e2b944c38a14b45580a23891a13d8b1155f2bd09 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Mon, 27 Mar 2023 17:22:49 +0200
Subject: [PATCH] rcw: Split up a050426.rcw to permit customized small PBIs

With, e.g., a combination of AN13022 and A-050426 and the usual other
things, we exceed the 4KB overall PBI size. To permit generating
smaller PBIs per board design needs, the original a050426.rcw has been
split up into sub-includes so that a user can copy it into the board
specific file and simply comment out the PEX or SD Ethernet files that
are not needed. This can substantially shrink the PBI size.

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 lx2160asi/a050426.rcw               | 338 +---------------------------
 lx2160asi/a050426_pex1.rcw          |  14 ++
 lx2160asi/a050426_pex2.rcw          |  14 ++
 lx2160asi/a050426_pex3.rcw          |  20 ++
 lx2160asi/a050426_pex4.rcw          |  14 ++
 lx2160asi/a050426_pex5.rcw          |  20 ++
 lx2160asi/a050426_pex6.rcw          |  14 ++
 lx2160asi/a050426_pex_prefix.rcw    |   9 +
 lx2160asi/a050426_post.rcw          |  21 ++
 lx2160asi/a050426_pre.rcw           |  38 ++++
 lx2160asi/a050426_sd1_eth.rcw       | 215 ++++++++++++++++++
 lx2160asi/a050426_sd2_eth.rcw       |  33 +++
 lx2160asi/a050426_sd_eth_prefix.rcw |   9 +
 13 files changed, 433 insertions(+), 326 deletions(-)
 create mode 100644 lx2160asi/a050426_pex1.rcw
 create mode 100644 lx2160asi/a050426_pex2.rcw
 create mode 100644 lx2160asi/a050426_pex3.rcw
 create mode 100644 lx2160asi/a050426_pex4.rcw
 create mode 100644 lx2160asi/a050426_pex5.rcw
 create mode 100644 lx2160asi/a050426_pex6.rcw
 create mode 100644 lx2160asi/a050426_pex_prefix.rcw
 create mode 100644 lx2160asi/a050426_post.rcw
 create mode 100644 lx2160asi/a050426_pre.rcw
 create mode 100644 lx2160asi/a050426_sd1_eth.rcw
 create mode 100644 lx2160asi/a050426_sd2_eth.rcw
 create mode 100644 lx2160asi/a050426_sd_eth_prefix.rcw

diff --git a/lx2160asi/a050426.rcw b/lx2160asi/a050426.rcw
index 191e4fa..6e8f8c7 100644
--- a/lx2160asi/a050426.rcw
+++ b/lx2160asi/a050426.rcw
@@ -10,329 +10,15 @@
  * and when the SoC comes out of POR reset.
  */
 
-.pbi
-blockcopy 0x0,0x01ea10a0,0x01e00210,0x4
-blockcopy 0x0,0x01ea10b0,0x01e00214,0xc
-blockcopy 0x0,0x01eb10a0,0x01e00220,0x4
-blockcopy 0x0,0x01eb10b0,0x01e00224,0xc
-write 0x1ea10a0, 0x0
-write 0x1ea10b0, 0x0
-write 0x1ea10b4, 0x0
-write 0x1ea10b8, 0x0
-write 0x1eb10a0, 0x0
-write 0x1eb10b0, 0x0
-write 0x1eb10b4, 0x0
-write 0x1eb10b8, 0x0
-.end
-
-.pbi
-loadacwindow 0x1C0
-awrite 0x117E60,0x80000001
-.end
-
-/* PEX */
-.pbi
-loadacwindow 0x1C2
-awrite.b4 0x2508000,0x64a8150e,0xcfc4885c
-awrite 0x2508008,0x40ee75fe
-awrite.b5 0x2518000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2518010,0x4416e4be
-awrite.b5 0x2520000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2520010,0x4416e4be
-awrite.b4 0x2608000,0x64a8150e,0xcfc4885c
-awrite 0x2608008,0x40ee75fe
-awrite.b5 0x2618000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2618010,0x4416e4be
-awrite.b5 0x2620000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2620010,0x4416e4be
-awrite.b4 0x2808000,0x64a8150e,0xcfc4885c
-awrite 0x2808008,0x40ee75fe
-awrite.b5 0x2818000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2818010,0x4416e4be
-awrite.b5 0x2820000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2820010,0x4416e4be
-awrite.b4 0x2B08000,0x64a8150e,0xcfc4885c
-awrite 0x2B08008,0x40ee75fe
-awrite.b5 0x2B18000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2B18010,0x4416e4be
-awrite.b5 0x2B20000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2B20010,0x4416e4be
-awrite.b5 0x2708000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2708010,0x4416e4be
-awrite.b5 0x2728000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2728010,0x4416e4be
-awrite.b5 0x2730000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2730010,0x4416e4be
-awrite.b5 0x2738000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2738010,0x4416e4be
-awrite.b5 0x2748000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2748010,0x4416e4be
-awrite.b5 0x2758000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2758010,0x4416e4be
-awrite.b5 0x2A08000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2A08010,0x4416e4be
-awrite.b5 0x2A28000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2A28010,0x4416e4be
-awrite.b5 0x2A30000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2A30010,0x4416e4be
-awrite.b5 0x2A38000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2A38010,0x4416e4be
-awrite.b5 0x2A48000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2A48010,0x4416e4be
-awrite.b5 0x2A58000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x2A58010,0x4416e4be
-.end
-
-/* lnx1_e100g */
-.pbi
-loadacwindow 0x1C3
-awrite.b4 0xA000,0x64a8150e,0xcfc4885c
-awrite 0xA008,0x40ee75fe
-awrite.b4 0xA200,0x64a8150e,0xcfc4885c
-awrite 0xA208,0x40ee75fe
-awrite.b4 0xA400,0x64a8150e,0xcfc4885c
-awrite 0xA408,0x40ee75fe
-awrite.b4 0xA600,0x64a8150e,0xcfc4885c
-awrite 0xA608,0x40ee75fe
-awrite.b4 0xA800,0x64a8150e,0xcfc4885c
-awrite 0xA808,0x40ee75fe
-awrite.b4 0xAA00,0x64a8150e,0xcfc4885c
-awrite 0xAA08,0x40ee75fe
-awrite.b4 0xAC00,0x64a8150e,0xcfc4885c
-awrite 0xAC08,0x40ee75fe
-awrite.b4 0xAE00,0x64a8150e,0xcfc4885c
-awrite 0xAE08,0x40ee75fe
-awrite.b4 0xB000,0x64a8150e,0xcfc4885c
-awrite 0xB008,0x40ee75fe
-awrite.b4 0xB200,0x64a8150e,0xcfc4885c
-awrite 0xB208,0x40ee75fe
-awrite.b4 0xB400,0x64a8150e,0xcfc4885c
-awrite 0xB408,0x40ee75fe
-awrite.b4 0xB600,0x64a8150e,0xcfc4885c
-awrite 0xB608,0x40ee75fe
-awrite.b4 0xB800,0x64a8150e,0xcfc4885c
-awrite 0xB808,0x40ee75fe
-awrite.b4 0xBA00,0x64a8150e,0xcfc4885c
-awrite 0xBA08,0x40ee75fe
-awrite.b4 0xBC00,0x64a8150e,0xcfc4885c
-awrite 0xBC08,0x40ee75fe
-awrite.b4 0xBE00,0x64a8150e,0xcfc4885c
-awrite 0xBE08,0x40ee75fe
-awrite.b5 0xC000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xC010,0x4416e4be
-awrite.b5 0xC400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xC410,0x4416e4be
-awrite.b5 0xC800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xC810,0x4416e4be
-awrite.b5 0xCC00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xCC10,0x4416e4be
-awrite.b5 0xD000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xD010,0x4416e4be
-awrite.b5 0xD400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xD410,0x4416e4be
-awrite.b5 0xD800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xD810,0x4416e4be
-awrite.b5 0xDC00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0xDC10,0x4416e4be
-awrite.b4 0xE000,0x64a8150e,0xcfc4885c
-awrite 0xE008,0x40ee75fe
-awrite.b4 0xF000,0x64a8150e,0xcfc4885c
-awrite 0xF008,0x40ee75fe
-awrite.b4 0x12000,0x64a8150e,0xcfc4885c
-awrite 0x12008,0x40ee75fe
-awrite.b4 0x12200,0x64a8150e,0xcfc4885c
-awrite 0x12208,0x40ee75fe
-awrite.b4 0x12400,0x64a8150e,0xcfc4885c
-awrite 0x12408,0x40ee75fe
-awrite.b4 0x12600,0x64a8150e,0xcfc4885c
-awrite 0x12608,0x40ee75fe
-awrite.b4 0x12800,0x64a8150e,0xcfc4885c
-awrite 0x12808,0x40ee75fe
-awrite.b4 0x12A00,0x64a8150e,0xcfc4885c
-awrite 0x12A08,0x40ee75fe
-awrite.b4 0x12C00,0x64a8150e,0xcfc4885c
-awrite 0x12C08,0x40ee75fe
-awrite.b4 0x12E00,0x64a8150e,0xcfc4885c
-awrite 0x12E08,0x40ee75fe
-awrite.b4 0x13000,0x64a8150e,0xcfc4885c
-awrite 0x13008,0x40ee75fe
-awrite.b4 0x13200,0x64a8150e,0xcfc4885c
-awrite 0x13208,0x40ee75fe
-awrite.b4 0x13400,0x64a8150e,0xcfc4885c
-awrite 0x13408,0x40ee75fe
-awrite.b4 0x13600,0x64a8150e,0xcfc4885c
-awrite 0x13608,0x40ee75fe
-awrite.b4 0x13800,0x64a8150e,0xcfc4885c
-awrite 0x13808,0x40ee75fe
-awrite.b4 0x13A00,0x64a8150e,0xcfc4885c
-awrite 0x13A08,0x40ee75fe
-awrite.b4 0x13C00,0x64a8150e,0xcfc4885c
-awrite 0x13C08,0x40ee75fe
-awrite.b4 0x13E00,0x64a8150e,0xcfc4885c
-awrite 0x13E08,0x40ee75fe
-awrite.b5 0x14000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x14010,0x4416e4be
-awrite.b5 0x14400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x14410,0x4416e4be
-awrite.b5 0x14800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x14810,0x4416e4be
-awrite.b5 0x14C00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x14C10,0x4416e4be
-awrite.b5 0x15000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x15010,0x4416e4be
-awrite.b5 0x15400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x15410,0x4416e4be
-awrite.b5 0x15800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x15810,0x4416e4be
-awrite.b5 0x15C00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x15C10,0x4416e4be
-awrite.b4 0x16000,0x64a8150e,0xcfc4885c
-awrite 0x16008,0x40ee75fe
-awrite.b4 0x17000,0x64a8150e,0xcfc4885c
-awrite 0x17008,0x40ee75fe
-.end
-
-/* lnx1_xfi */
-.pbi
-awrite.b4 0x108000,0x64a8150e,0xcfc4885c
-awrite 0x108008,0x40ee75fe
-awrite.b4 0x108200,0x64a8150e,0xcfc4885c
-awrite 0x108208,0x40ee75fe
-awrite.b4 0x10A000,0x64a8150e,0xcfc4885c
-awrite 0x10A008,0x40ee75fe
-awrite.b4 0x10A400,0x64a8150e,0xcfc4885c
-awrite 0x10A408,0x40ee75fe
-awrite.b5 0x10C000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x10C010,0x4416e4be
-awrite.b5 0x10C400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x10C410,0x4416e4be
-awrite.b4 0x10E000,0x64a8150e,0xcfc4885c
-awrite 0x10E008,0x40ee75fe
-awrite.b4 0x10E200,0x64a8150e,0xcfc4885c
-awrite 0x10E208,0x40ee75fe
-awrite.b4 0x110000,0x64a8150e,0xcfc4885c
-awrite 0x110008,0x40ee75fe
-awrite.b4 0x110400,0x64a8150e,0xcfc4885c
-awrite 0x110408,0x40ee75fe
-awrite.b5 0x112000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x112010,0x4416e4be
-awrite.b5 0x112400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x112410,0x4416e4be
-awrite.b4 0x114000,0x64a8150e,0xcfc4885c
-awrite 0x114008,0x40ee75fe
-awrite.b4 0x114200,0x64a8150e,0xcfc4885c
-awrite 0x114208,0x40ee75fe
-awrite.b4 0x116000,0x64a8150e,0xcfc4885c
-awrite 0x116008,0x40ee75fe
-awrite.b4 0x116400,0x64a8150e,0xcfc4885c
-awrite 0x116408,0x40ee75fe
-awrite.b5 0x118000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x118010,0x4416e4be
-awrite.b5 0x118400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x118410,0x4416e4be
-awrite.b4 0x11A000,0x64a8150e,0xcfc4885c
-awrite 0x11A008,0x40ee75fe
-awrite.b4 0x11A200,0x64a8150e,0xcfc4885c
-awrite 0x11A208,0x40ee75fe
-awrite.b4 0x11C000,0x64a8150e,0xcfc4885c
-awrite 0x11C008,0x40ee75fe
-awrite.b4 0x11C400,0x64a8150e,0xcfc4885c
-awrite 0x11C408,0x40ee75fe
-awrite.b5 0x11E000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x11E010,0x4416e4be
-awrite.b5 0x11E400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x11E410,0x4416e4be
-awrite.b4 0x120000,0x64a8150e,0xcfc4885c
-awrite 0x120008,0x40ee75fe
-awrite.b4 0x120200,0x64a8150e,0xcfc4885c
-awrite 0x120208,0x40ee75fe
-awrite.b4 0x122000,0x64a8150e,0xcfc4885c
-awrite 0x122008,0x40ee75fe
-awrite.b4 0x122400,0x64a8150e,0xcfc4885c
-awrite 0x122408,0x40ee75fe
-awrite.b5 0x124000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x124010,0x4416e4be
-awrite.b5 0x124400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x124410,0x4416e4be
-awrite.b4 0x126000,0x64a8150e,0xcfc4885c
-awrite 0x126008,0x40ee75fe
-awrite.b4 0x126200,0x64a8150e,0xcfc4885c
-awrite 0x126208,0x40ee75fe
-awrite.b4 0x128000,0x64a8150e,0xcfc4885c
-awrite 0x128008,0x40ee75fe
-awrite.b4 0x128400,0x64a8150e,0xcfc4885c
-awrite 0x128408,0x40ee75fe
-awrite.b5 0x12A000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x12A010,0x4416e4be
-awrite.b5 0x12A400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x12A410,0x4416e4be
-awrite.b4 0x12C000,0x64a8150e,0xcfc4885c
-awrite 0x12C008,0x40ee75fe
-awrite.b4 0x12C200,0x64a8150e,0xcfc4885c
-awrite 0x12C208,0x40ee75fe
-awrite.b4 0x12E000,0x64a8150e,0xcfc4885c
-awrite 0x12E008,0x40ee75fe
-awrite.b4 0x12E400,0x64a8150e,0xcfc4885c
-awrite 0x12E408,0x40ee75fe
-awrite.b5 0x130000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x130010,0x4416e4be
-awrite.b5 0x130400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x130410,0x4416e4be
-awrite.b4 0x132000,0x64a8150e,0xcfc4885c
-awrite 0x132008,0x40ee75fe
-awrite.b4 0x132200,0x64a8150e,0xcfc4885c
-awrite 0x132208,0x40ee75fe
-awrite.b4 0x134000,0x64a8150e,0xcfc4885c
-awrite 0x134008,0x40ee75fe
-awrite.b4 0x134400,0x64a8150e,0xcfc4885c
-awrite 0x134408,0x40ee75fe
-awrite.b5 0x136000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x136010,0x4416e4be
-awrite.b5 0x136400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x136410,0x4416e4be
-.end
-
-/* lnx2_xfi */
-.pbi
-awrite.b4 0x308000,0x64a8150e,0xcfc4885c
-awrite 0x308008,0x40ee75fe
-awrite.b4 0x308200,0x64a8150e,0xcfc4885c
-awrite 0x308208,0x40ee75fe
-awrite.b4 0x30A000,0x64a8150e,0xcfc4885c
-awrite 0x30A008,0x40ee75fe
-awrite.b4 0x30A400,0x64a8150e,0xcfc4885c
-awrite 0x30A408,0x40ee75fe
-awrite.b5 0x30C000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x30C010,0x4416e4be
-awrite.b5 0x30C400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x30C410,0x4416e4be
-awrite.b4 0x30E000,0x64a8150e,0xcfc4885c
-awrite 0x30E008,0x40ee75fe
-awrite.b4 0x30E200,0x64a8150e,0xcfc4885c
-awrite 0x30E208,0x40ee75fe
-awrite.b4 0x310000,0x64a8150e,0xcfc4885c
-awrite 0x310008,0x40ee75fe
-awrite.b4 0x310400,0x64a8150e,0xcfc4885c
-awrite 0x310408,0x40ee75fe
-awrite.b5 0x312000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x312010,0x4416e4be
-awrite.b5 0x312400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
-awrite 0x312410,0x4416e4be
-.end
-
-.pbi
-loadacwindow 0x1C0
-awrite 0x117E60,0x0
-.end
-
-.pbi
-blockcopy 0x0,0x01e00210,0x01ea10a0,0x4
-blockcopy 0x0,0x01e00214,0x01ea10b0,0xc
-blockcopy 0x0,0x01e00220,0x01eb10a0,0x4
-blockcopy 0x0,0x01e00224,0x01eb10b0,0xc
-.end
-
-/* Write a test value to SCRATCHRW5 for validation purpose */
-.pbi
-write 0x01e00210,0x00050426
-.end
+#include <../lx2160asi/a050426_pre.rcw>
+#include <../lx2160asi/a050426_pex_prefix.rcw>
+#include <../lx2160asi/a050426_pex1.rcw>
+#include <../lx2160asi/a050426_pex2.rcw>
+#include <../lx2160asi/a050426_pex3.rcw>
+#include <../lx2160asi/a050426_pex4.rcw>
+#include <../lx2160asi/a050426_pex5.rcw>
+#include <../lx2160asi/a050426_pex6.rcw>
+#include <../lx2160asi/a050426_sd_eth_prefix.rcw>
+#include <../lx2160asi/a050426_sd1_eth.rcw>
+#include <../lx2160asi/a050426_sd2_eth.rcw>
+#include <../lx2160asi/a050426_post.rcw>
diff --git a/lx2160asi/a050426_pex1.rcw b/lx2160asi/a050426_pex1.rcw
new file mode 100644
index 0000000..d75902a
--- /dev/null
+++ b/lx2160asi/a050426_pex1.rcw
@@ -0,0 +1,14 @@
+/*
+ * This must be included after a050426_pex_prefix.rcw and before
+ * a050426_post.rcw if PEX1 is not disabled.
+ */
+
+/* PEX 1 */
+.pbi
+awrite.b4 0x2508000,0x64a8150e,0xcfc4885c
+awrite 0x2508008,0x40ee75fe
+awrite.b5 0x2518000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2518010,0x4416e4be
+awrite.b5 0x2520000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2520010,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_pex2.rcw b/lx2160asi/a050426_pex2.rcw
new file mode 100644
index 0000000..cfe6ef1
--- /dev/null
+++ b/lx2160asi/a050426_pex2.rcw
@@ -0,0 +1,14 @@
+/*
+ * This must be included after a050426_pex_prefix.rcw and before
+ * a050426_post.rcw if PEX2 is not disabled.
+ */
+
+/* PEX 2 */
+.pbi
+awrite.b4 0x2608000,0x64a8150e,0xcfc4885c
+awrite 0x2608008,0x40ee75fe
+awrite.b5 0x2618000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2618010,0x4416e4be
+awrite.b5 0x2620000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2620010,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_pex3.rcw b/lx2160asi/a050426_pex3.rcw
new file mode 100644
index 0000000..e5b7c24
--- /dev/null
+++ b/lx2160asi/a050426_pex3.rcw
@@ -0,0 +1,20 @@
+/*
+ * This must be included after a050426_pex_prefix.rcw and before
+ * a050426_post.rcw if PEX3 is not disabled.
+ */
+
+/* PEX 3 */
+.pbi
+awrite.b5 0x2708000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2708010,0x4416e4be
+awrite.b5 0x2728000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2728010,0x4416e4be
+awrite.b5 0x2730000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2730010,0x4416e4be
+awrite.b5 0x2738000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2738010,0x4416e4be
+awrite.b5 0x2748000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2748010,0x4416e4be
+awrite.b5 0x2758000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2758010,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_pex4.rcw b/lx2160asi/a050426_pex4.rcw
new file mode 100644
index 0000000..d8c512a
--- /dev/null
+++ b/lx2160asi/a050426_pex4.rcw
@@ -0,0 +1,14 @@
+/*
+ * This must be included after a050426_pex_prefix.rcw and before
+ * a050426_post.rcw if PEX4 is not disabled.
+ */
+
+/* PEX 4 */
+.pbi
+awrite.b4 0x2808000,0x64a8150e,0xcfc4885c
+awrite 0x2808008,0x40ee75fe
+awrite.b5 0x2818000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2818010,0x4416e4be
+awrite.b5 0x2820000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2820010,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_pex5.rcw b/lx2160asi/a050426_pex5.rcw
new file mode 100644
index 0000000..757d024
--- /dev/null
+++ b/lx2160asi/a050426_pex5.rcw
@@ -0,0 +1,20 @@
+/*
+ * This must be included after a050426_pex_prefix.rcw and before
+ * a050426_post.rcw if PEX5 is not disabled.
+ */
+
+/* PEX 5 */
+.pbi
+awrite.b5 0x2A08000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2A08010,0x4416e4be
+awrite.b5 0x2A28000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2A28010,0x4416e4be
+awrite.b5 0x2A30000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2A30010,0x4416e4be
+awrite.b5 0x2A38000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2A38010,0x4416e4be
+awrite.b5 0x2A48000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2A48010,0x4416e4be
+awrite.b5 0x2A58000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2A58010,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_pex6.rcw b/lx2160asi/a050426_pex6.rcw
new file mode 100644
index 0000000..17602d8
--- /dev/null
+++ b/lx2160asi/a050426_pex6.rcw
@@ -0,0 +1,14 @@
+/*
+ * This must be included after a050426_pex_prefix.rcw and before
+ * a050426_post.rcw if PEX6 is not disabled.
+ */
+
+/* PEX 6 */
+.pbi
+awrite.b4 0x2B08000,0x64a8150e,0xcfc4885c
+awrite 0x2B08008,0x40ee75fe
+awrite.b5 0x2B18000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2B18010,0x4416e4be
+awrite.b5 0x2B20000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x2B20010,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_pex_prefix.rcw b/lx2160asi/a050426_pex_prefix.rcw
new file mode 100644
index 0000000..35ee8e1
--- /dev/null
+++ b/lx2160asi/a050426_pex_prefix.rcw
@@ -0,0 +1,9 @@
+/*
+ * This must be included after a050426_pre.rcw and immediately before
+ * any a050426_pex?.rcw.
+ */
+
+/* PEX */
+.pbi
+loadacwindow 0x1C2
+.end
diff --git a/lx2160asi/a050426_post.rcw b/lx2160asi/a050426_post.rcw
new file mode 100644
index 0000000..6703e8e
--- /dev/null
+++ b/lx2160asi/a050426_post.rcw
@@ -0,0 +1,21 @@
+/*
+ * This must be included after a050426_pre.rcw and any port specific
+ * a050426_*.rcw includes.
+ */
+
+.pbi
+loadacwindow 0x1C0
+awrite 0x117E60,0x0
+.end
+
+.pbi
+blockcopy 0x0,0x01e00210,0x01ea10a0,0x4
+blockcopy 0x0,0x01e00214,0x01ea10b0,0xc
+blockcopy 0x0,0x01e00220,0x01eb10a0,0x4
+blockcopy 0x0,0x01e00224,0x01eb10b0,0xc
+.end
+
+/* Write a test value to SCRATCHRW5 for validation purpose */
+.pbi
+write 0x01e00210,0x00050426
+.end
diff --git a/lx2160asi/a050426_pre.rcw b/lx2160asi/a050426_pre.rcw
new file mode 100644
index 0000000..da8951d
--- /dev/null
+++ b/lx2160asi/a050426_pre.rcw
@@ -0,0 +1,38 @@
+/*
+ * Work-around for erratum A-050426
+ *
+ * Problem: After PORESET de-assertion, internal RAM first write or
+ * read access corruption may occur
+ * The internal RAM associated with QDMA, FECA, and PCIe may fail
+ * the first memory write or read access after completion of reset
+ *
+ * Workaround: PBI sequence is required after the boot sequence is completed
+ * and when the SoC comes out of POR reset.
+ */
+
+/* NOTE: This is only the start file of the PBI fix! */
+
+.pbi
+blockcopy 0x0,0x01ea10a0,0x01e00210,0x4
+blockcopy 0x0,0x01ea10b0,0x01e00214,0xc
+blockcopy 0x0,0x01eb10a0,0x01e00220,0x4
+blockcopy 0x0,0x01eb10b0,0x01e00224,0xc
+write 0x1ea10a0, 0x0
+write 0x1ea10b0, 0x0
+write 0x1ea10b4, 0x0
+write 0x1ea10b8, 0x0
+write 0x1eb10a0, 0x0
+write 0x1eb10b0, 0x0
+write 0x1eb10b4, 0x0
+write 0x1eb10b8, 0x0
+.end
+
+.pbi
+loadacwindow 0x1C0
+awrite 0x117E60,0x80000001
+.end
+
+/*
+ * Now the port specific includes need to happen and a final include
+ * of a050426_post.rcw
+ */
diff --git a/lx2160asi/a050426_sd1_eth.rcw b/lx2160asi/a050426_sd1_eth.rcw
new file mode 100644
index 0000000..b0f597e
--- /dev/null
+++ b/lx2160asi/a050426_sd1_eth.rcw
@@ -0,0 +1,215 @@
+/*
+ * This must be included after a050426_sd_prefix.rcw and before
+ * a050426_post.rcw if any Ethernet is used on SerDes 1
+ */
+
+/* SerDes 1 */
+/* lnx1_e100g */
+.pbi
+loadacwindow 0x1C3
+awrite.b4 0xA000,0x64a8150e,0xcfc4885c
+awrite 0xA008,0x40ee75fe
+awrite.b4 0xA200,0x64a8150e,0xcfc4885c
+awrite 0xA208,0x40ee75fe
+awrite.b4 0xA400,0x64a8150e,0xcfc4885c
+awrite 0xA408,0x40ee75fe
+awrite.b4 0xA600,0x64a8150e,0xcfc4885c
+awrite 0xA608,0x40ee75fe
+awrite.b4 0xA800,0x64a8150e,0xcfc4885c
+awrite 0xA808,0x40ee75fe
+awrite.b4 0xAA00,0x64a8150e,0xcfc4885c
+awrite 0xAA08,0x40ee75fe
+awrite.b4 0xAC00,0x64a8150e,0xcfc4885c
+awrite 0xAC08,0x40ee75fe
+awrite.b4 0xAE00,0x64a8150e,0xcfc4885c
+awrite 0xAE08,0x40ee75fe
+awrite.b4 0xB000,0x64a8150e,0xcfc4885c
+awrite 0xB008,0x40ee75fe
+awrite.b4 0xB200,0x64a8150e,0xcfc4885c
+awrite 0xB208,0x40ee75fe
+awrite.b4 0xB400,0x64a8150e,0xcfc4885c
+awrite 0xB408,0x40ee75fe
+awrite.b4 0xB600,0x64a8150e,0xcfc4885c
+awrite 0xB608,0x40ee75fe
+awrite.b4 0xB800,0x64a8150e,0xcfc4885c
+awrite 0xB808,0x40ee75fe
+awrite.b4 0xBA00,0x64a8150e,0xcfc4885c
+awrite 0xBA08,0x40ee75fe
+awrite.b4 0xBC00,0x64a8150e,0xcfc4885c
+awrite 0xBC08,0x40ee75fe
+awrite.b4 0xBE00,0x64a8150e,0xcfc4885c
+awrite 0xBE08,0x40ee75fe
+awrite.b5 0xC000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xC010,0x4416e4be
+awrite.b5 0xC400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xC410,0x4416e4be
+awrite.b5 0xC800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xC810,0x4416e4be
+awrite.b5 0xCC00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xCC10,0x4416e4be
+awrite.b5 0xD000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xD010,0x4416e4be
+awrite.b5 0xD400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xD410,0x4416e4be
+awrite.b5 0xD800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xD810,0x4416e4be
+awrite.b5 0xDC00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0xDC10,0x4416e4be
+awrite.b4 0xE000,0x64a8150e,0xcfc4885c
+awrite 0xE008,0x40ee75fe
+awrite.b4 0xF000,0x64a8150e,0xcfc4885c
+awrite 0xF008,0x40ee75fe
+awrite.b4 0x12000,0x64a8150e,0xcfc4885c
+awrite 0x12008,0x40ee75fe
+awrite.b4 0x12200,0x64a8150e,0xcfc4885c
+awrite 0x12208,0x40ee75fe
+awrite.b4 0x12400,0x64a8150e,0xcfc4885c
+awrite 0x12408,0x40ee75fe
+awrite.b4 0x12600,0x64a8150e,0xcfc4885c
+awrite 0x12608,0x40ee75fe
+awrite.b4 0x12800,0x64a8150e,0xcfc4885c
+awrite 0x12808,0x40ee75fe
+awrite.b4 0x12A00,0x64a8150e,0xcfc4885c
+awrite 0x12A08,0x40ee75fe
+awrite.b4 0x12C00,0x64a8150e,0xcfc4885c
+awrite 0x12C08,0x40ee75fe
+awrite.b4 0x12E00,0x64a8150e,0xcfc4885c
+awrite 0x12E08,0x40ee75fe
+awrite.b4 0x13000,0x64a8150e,0xcfc4885c
+awrite 0x13008,0x40ee75fe
+awrite.b4 0x13200,0x64a8150e,0xcfc4885c
+awrite 0x13208,0x40ee75fe
+awrite.b4 0x13400,0x64a8150e,0xcfc4885c
+awrite 0x13408,0x40ee75fe
+awrite.b4 0x13600,0x64a8150e,0xcfc4885c
+awrite 0x13608,0x40ee75fe
+awrite.b4 0x13800,0x64a8150e,0xcfc4885c
+awrite 0x13808,0x40ee75fe
+awrite.b4 0x13A00,0x64a8150e,0xcfc4885c
+awrite 0x13A08,0x40ee75fe
+awrite.b4 0x13C00,0x64a8150e,0xcfc4885c
+awrite 0x13C08,0x40ee75fe
+awrite.b4 0x13E00,0x64a8150e,0xcfc4885c
+awrite 0x13E08,0x40ee75fe
+awrite.b5 0x14000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x14010,0x4416e4be
+awrite.b5 0x14400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x14410,0x4416e4be
+awrite.b5 0x14800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x14810,0x4416e4be
+awrite.b5 0x14C00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x14C10,0x4416e4be
+awrite.b5 0x15000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x15010,0x4416e4be
+awrite.b5 0x15400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x15410,0x4416e4be
+awrite.b5 0x15800,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x15810,0x4416e4be
+awrite.b5 0x15C00,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x15C10,0x4416e4be
+awrite.b4 0x16000,0x64a8150e,0xcfc4885c
+awrite 0x16008,0x40ee75fe
+awrite.b4 0x17000,0x64a8150e,0xcfc4885c
+awrite 0x17008,0x40ee75fe
+.end
+
+/* SerDes 1 */
+/* lnx1_xfi */
+.pbi
+awrite.b4 0x108000,0x64a8150e,0xcfc4885c
+awrite 0x108008,0x40ee75fe
+awrite.b4 0x108200,0x64a8150e,0xcfc4885c
+awrite 0x108208,0x40ee75fe
+awrite.b4 0x10A000,0x64a8150e,0xcfc4885c
+awrite 0x10A008,0x40ee75fe
+awrite.b4 0x10A400,0x64a8150e,0xcfc4885c
+awrite 0x10A408,0x40ee75fe
+awrite.b5 0x10C000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x10C010,0x4416e4be
+awrite.b5 0x10C400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x10C410,0x4416e4be
+awrite.b4 0x10E000,0x64a8150e,0xcfc4885c
+awrite 0x10E008,0x40ee75fe
+awrite.b4 0x10E200,0x64a8150e,0xcfc4885c
+awrite 0x10E208,0x40ee75fe
+awrite.b4 0x110000,0x64a8150e,0xcfc4885c
+awrite 0x110008,0x40ee75fe
+awrite.b4 0x110400,0x64a8150e,0xcfc4885c
+awrite 0x110408,0x40ee75fe
+awrite.b5 0x112000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x112010,0x4416e4be
+awrite.b5 0x112400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x112410,0x4416e4be
+awrite.b4 0x114000,0x64a8150e,0xcfc4885c
+awrite 0x114008,0x40ee75fe
+awrite.b4 0x114200,0x64a8150e,0xcfc4885c
+awrite 0x114208,0x40ee75fe
+awrite.b4 0x116000,0x64a8150e,0xcfc4885c
+awrite 0x116008,0x40ee75fe
+awrite.b4 0x116400,0x64a8150e,0xcfc4885c
+awrite 0x116408,0x40ee75fe
+awrite.b5 0x118000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x118010,0x4416e4be
+awrite.b5 0x118400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x118410,0x4416e4be
+awrite.b4 0x11A000,0x64a8150e,0xcfc4885c
+awrite 0x11A008,0x40ee75fe
+awrite.b4 0x11A200,0x64a8150e,0xcfc4885c
+awrite 0x11A208,0x40ee75fe
+awrite.b4 0x11C000,0x64a8150e,0xcfc4885c
+awrite 0x11C008,0x40ee75fe
+awrite.b4 0x11C400,0x64a8150e,0xcfc4885c
+awrite 0x11C408,0x40ee75fe
+awrite.b5 0x11E000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x11E010,0x4416e4be
+awrite.b5 0x11E400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x11E410,0x4416e4be
+awrite.b4 0x120000,0x64a8150e,0xcfc4885c
+awrite 0x120008,0x40ee75fe
+awrite.b4 0x120200,0x64a8150e,0xcfc4885c
+awrite 0x120208,0x40ee75fe
+awrite.b4 0x122000,0x64a8150e,0xcfc4885c
+awrite 0x122008,0x40ee75fe
+awrite.b4 0x122400,0x64a8150e,0xcfc4885c
+awrite 0x122408,0x40ee75fe
+awrite.b5 0x124000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x124010,0x4416e4be
+awrite.b5 0x124400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x124410,0x4416e4be
+awrite.b4 0x126000,0x64a8150e,0xcfc4885c
+awrite 0x126008,0x40ee75fe
+awrite.b4 0x126200,0x64a8150e,0xcfc4885c
+awrite 0x126208,0x40ee75fe
+awrite.b4 0x128000,0x64a8150e,0xcfc4885c
+awrite 0x128008,0x40ee75fe
+awrite.b4 0x128400,0x64a8150e,0xcfc4885c
+awrite 0x128408,0x40ee75fe
+awrite.b5 0x12A000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x12A010,0x4416e4be
+awrite.b5 0x12A400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x12A410,0x4416e4be
+awrite.b4 0x12C000,0x64a8150e,0xcfc4885c
+awrite 0x12C008,0x40ee75fe
+awrite.b4 0x12C200,0x64a8150e,0xcfc4885c
+awrite 0x12C208,0x40ee75fe
+awrite.b4 0x12E000,0x64a8150e,0xcfc4885c
+awrite 0x12E008,0x40ee75fe
+awrite.b4 0x12E400,0x64a8150e,0xcfc4885c
+awrite 0x12E408,0x40ee75fe
+awrite.b5 0x130000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x130010,0x4416e4be
+awrite.b5 0x130400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x130410,0x4416e4be
+awrite.b4 0x132000,0x64a8150e,0xcfc4885c
+awrite 0x132008,0x40ee75fe
+awrite.b4 0x132200,0x64a8150e,0xcfc4885c
+awrite 0x132208,0x40ee75fe
+awrite.b4 0x134000,0x64a8150e,0xcfc4885c
+awrite 0x134008,0x40ee75fe
+awrite.b4 0x134400,0x64a8150e,0xcfc4885c
+awrite 0x134408,0x40ee75fe
+awrite.b5 0x136000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x136010,0x4416e4be
+awrite.b5 0x136400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x136410,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_sd2_eth.rcw b/lx2160asi/a050426_sd2_eth.rcw
new file mode 100644
index 0000000..bf84df7
--- /dev/null
+++ b/lx2160asi/a050426_sd2_eth.rcw
@@ -0,0 +1,33 @@
+/*
+ * This must be included after a050426_sd_prefix.rcw and before
+ * a050426_post.rcw if any Ethernet is used on SerDes 2
+ */
+
+/* SerDes 2 */
+/* lnx2_xfi */
+.pbi
+awrite.b4 0x308000,0x64a8150e,0xcfc4885c
+awrite 0x308008,0x40ee75fe
+awrite.b4 0x308200,0x64a8150e,0xcfc4885c
+awrite 0x308208,0x40ee75fe
+awrite.b4 0x30A000,0x64a8150e,0xcfc4885c
+awrite 0x30A008,0x40ee75fe
+awrite.b4 0x30A400,0x64a8150e,0xcfc4885c
+awrite 0x30A408,0x40ee75fe
+awrite.b5 0x30C000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x30C010,0x4416e4be
+awrite.b5 0x30C400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x30C410,0x4416e4be
+awrite.b4 0x30E000,0x64a8150e,0xcfc4885c
+awrite 0x30E008,0x40ee75fe
+awrite.b4 0x30E200,0x64a8150e,0xcfc4885c
+awrite 0x30E208,0x40ee75fe
+awrite.b4 0x310000,0x64a8150e,0xcfc4885c
+awrite 0x310008,0x40ee75fe
+awrite.b4 0x310400,0x64a8150e,0xcfc4885c
+awrite 0x310408,0x40ee75fe
+awrite.b5 0x312000,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x312010,0x4416e4be
+awrite.b5 0x312400,0x64a8150e,0xcfc4885c,0x40ee75fe,0x4b19116f
+awrite 0x312410,0x4416e4be
+.end
diff --git a/lx2160asi/a050426_sd_eth_prefix.rcw b/lx2160asi/a050426_sd_eth_prefix.rcw
new file mode 100644
index 0000000..0de03ff
--- /dev/null
+++ b/lx2160asi/a050426_sd_eth_prefix.rcw
@@ -0,0 +1,9 @@
+/*
+ * This must be included after a050426_pre.rcw and immediately before
+ * any a050426_sd?_eth.rcw.
+ */
+
+/* SerDes 1 and 2 */
+.pbi
+loadacwindow 0x1C3
+.end
-- 
2.34.1

